sdk/src/mock_bus.rs

```diff
@@ -18,28 +18,26 @@ pub struct PublishRecord<M: MessageBounds> {
     pub message: Arc<M>,
 }
 
-#[derive(Clone)]
 pub struct MockSubscription<M> {
-    messages: Arc<Mutex<VecDeque<(String, Arc<M>)>>>,
-    notify: Arc<Notify>,
-    deregister: Arc<dyn Fn() + Send + Sync>,
+    id: usize,
+    receiver: mpsc::UnboundedReceiver<(String, Arc<M>)>,
```

Comment:
This is a convention I tend to follow with tokio channels: qualify the names with the packages they came from (e.g. call this type `mpsc::UnboundedReceiver` rather than `UnboundedReceiver`). That makes it easier to tell what the semantics of the channel are at a glance (you can see that a sender is `oneshot::Sender` vs `watch::Sender`), especially in crates which use more than one type of channel.